### PR TITLE
Fortanix SGX target libunwind build process changes

### PR DESCRIPTION
**Pull request by AdrianCX**
Ticket: https://github.com/fortanix/rust-sgx/issues/174
LLVM related changes (merged): https://github.com/rust-lang/llvm-project/pull/57

Description: libunwind changes needed to run code in sgx environment via rust-sgx.

Target that uses this in rust: x86_64-fortanix-unknown-sgx.

Without this change, rust std for this toolchain is forced to use a precompiled library loaded via environment variable.

With this change we act the same as musl target.

### CODE REVIEW

**Comment by jethrogb** [comment]
Let's wait with this until after #72655 is merged. I don't think you can remove `build-x86_64-fortanix-unknown-sgx-toolchain.sh` then.

**Comment by nikomatsakis** [comment]
Blocked on #72655

**Comment by jethrogb** [comment]
LGTM

**Review by jethrogb — APPROVED** [review]
LGTM

File: src/ci/docker/dist-various-2/Dockerfile [hunk 1]
```diff
@@ -73,7 +73,7 @@ COPY dist-various-2/x86_64-fortanix-unknown-sgx-clang-wrap.sh /usr/bin/x86_64-fo
 RUN ln -s /usr/bin/x86_64-fortanix-unknown-sgx-clang-11 /usr/bin/x86_64-fortanix-unknown-sgx-clang++-11
 # We pass the commit id of the port of LLVM's libunwind to the build script.
 # Any update to the commit id here, should cause the container image to be re-built from this point on.
```
Comment (jethrogb) [inline src/ci/docker/dist-various-2/Dockerfile:75]: Comment is now obsolete

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**